operator/internal/controller/auto/update_controller.go

```diff
@@ -168,6 +192,36 @@ func (r *UpdateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 }
 
+func isWorkspaceReady(ws *autov1alpha1.Workspace) bool {
+	if ws == nil || ws.Generation != ws.Status.ObservedGeneration {
+		return false
+	}
+	return meta.IsStatusConditionTrue(ws.Status.Conditions, autov1alpha1.WorkspaceReady)
+}
+
+type workspaceReadyPredicate struct {
+	predicate.Funcs
+}
+
+func (workspaceReadyPredicate) Create(e event.CreateEvent) bool {
+	return isWorkspaceReady(e.Object.(*autov1alpha1.Workspace))
+}
```

Comment:
Just to get ahead of some of the lint errors we'll need to clean up -- same goes for Delete/Generic.

```suggestion
func (workspaceReadyPredicate) Create(_ event.CreateEvent) bool {
	return isWorkspaceReady(e.Object.(*autov1alpha1.Workspace))
}
```